gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/ConcatenatedGigawordDocuments.java

```diff
@@ -18,18 +26,112 @@
+    this.articleList = aArticleList;
+    this.annotated = true;
+  }
+  
   public static ConcatenatedGigawordDocuments fromGigwordGZippedFile(Path p) throws IOException {
     return new ConcatenatedGigawordDocuments(GZIPByteSource.fromCompressed(p.toFile())
             .asCharSource(StandardCharsets.UTF_8).read());
   }
 
+  public static ConcatenatedGigawordDocuments fromAnnotatedGigwordGZippedFile(Path p) throws IOException {
+    // Initialize Streams
+    try (InputStream fileInputStream = new FileInputStream(p.toString());
+         GZIPInputStream gzipInputStream = new GZIPInputStream(fileInputStream);
+         InputStreamReader inputStreamReader = new InputStreamReader(gzipInputStream);
+         BufferedReader br = new BufferedReader(inputStreamReader)) {
+
+        String sCurrentLine;
+    
+        // variables for start of a new document
+        Pattern GIGAWORD_DOC_ELEMENT_PATTERN = Pattern.compile("<DOC id=\"(.*?)\".*>");
+        String currentDocId = "";
+    
+        // variables for saving articles
+        boolean headlineStarted = false;
+        boolean textStarted = false;
+        String docText = "";
+        List<Article> articleList = new ArrayList<>();
+    
+    
+        // read file
+        while ((sCurrentLine = br.readLine()) != null) {
+    
+            sCurrentLine = sCurrentLine.trim();
+    
+            // extract document ID
+            if (sCurrentLine.startsWith("<DOC id=")) {
+                Matcher m = GIGAWORD_DOC_ELEMENT_PATTERN.matcher(sCurrentLine);
+                if (m.find()) {
+                    currentDocId = m.group(1);
+                }
+                else {
+                    throw new RuntimeException("Missing document ID on article");
+                }
+            }
+    
+            // create article when end of document (</TEXT>) is reached
+            if (sCurrentLine.equals("</TEXT>")) {
+                articleList.add(new Article(currentDocId, docText));
+                docText = "";
+                textStarted = false;
+            }
+    
+            if (sCurrentLine.equals("</HEADLINE>")) {
+                docText += "\n\n";
+                headlineStarted = false;
+            }
+    
+            // read lines from document text
+            if ((headlineStarted || textStarted) && !sCurrentLine.startsWith("<")) {
+                if (sCurrentLine.equals("")) {
+                    docText += "\n";
+                }
+                else if (sCurrentLine.startsWith("(")) {
+                    String parsedText = PennTreeUtils.toText(PennTreeUtils.parsePennTree(sCurrentLine));
+                    if (!parsedText.equals("null")) {
+                        docText += parsedText + " ";
+                    }
+                }
+                // sometimes there is one unannotated line in the document which can be used as plain text
+                else {
+                    docText += sCurrentLine + " ";
```

Comment:
Why add a space at the end of the line?

Reply:
Because every line contains a sentence and there should be a space between the full stop of one sentence and the next sentence.

Reply:
If every line has a sentence, then we already have a line-break between the full stop and the next sentence, right? I don't think it is necessary to add an additional space character before/after the line break.

Reply:
There is no line-break between the full stop and the next sentence currently. I can do that, but wouldn't I just replace the `+ " "` with `+ "\n"` then?

Reply:
Currently line-breaks are only inserted if there is an empty line between to sentence-annotations. Which is only the case between paragraphs and not always.